code/modules/fish/fish_tank.dm

```diff
@@ -0,0 +1,685 @@
+		to_chat(user, "There are no fish in \the [src] to catch!")
+		return
+
+	var/caught_fish = input("Select a fish to catch.", "Fishing") as null|anything in fish_list		//Select a fish from the tank
+	if(caught_fish)
+		var/dead_fish = null
+		dead_fish = fish_items_list[caught_fish]	//Locate the appropriate fish_item for the caught fish
+		if(!dead_fish)								//No fish_item found, possibly due to typo or not being listed. Do nothing.
+			return
+		kill_fish(caught_fish)						//Kill the caught fish from the tank
+		user.visible_message("[user.name] harvests \a [caught_fish] from \the [src].", "You scoop \a [caught_fish] out of \the [src].")
+		new dead_fish(get_turf(user))				//Spawn the appropriate fish_item at the user's feet.
+
+/obj/machinery/fishtank/proc/destroy(var/deconstruct = FALSE)
+	if(!deconstruct)
+		to_chat(world, "Brutally")														//Check if we are deconstructing or breaking the tank
+		for(var/i in  shard_count)														//Produce the appropriate number of glass shards
+			var/obj/item/weapon/shard/S = new /obj/item/weapon/shard
+			S.forceMove(get_turf(src))
+		if(water_level)																	//Spill any water that was left in the tank when it broke
+			spill_water()
+	else																				//We are deconstructing, make glass sheets instead of shards
+		var/sheets = shard_count + 1													//Deconstructing it salvages all the glass used to build the tank
+		var/obj/item/stack/sheet/glass/stack = new /obj/item/stack/sheet/glass(sheets)	//Produce the appropriate number of glass sheets, in a single stack
+		stack.forceMove(get_turf(src))
+	qdel(src)																			//qdel the tank and it's contents
+
+
+/obj/machinery/fishtank/proc/spill_water()
+	switch(tank_type)
+		if("bowl")										//Fishbowl: Wets it's own tile
+			var/turf/T = get_turf(src)
+			if(!istype(T, /turf/simulated))
+				return
+			var/turf/simulated/S = T
+			S.wet = TRUE
+
+		if("tank")										//Fishtank: Wets it's own tile and the 4 adjacent tiles (cardinal directions)
+			var/turf/ST = get_turf(src)
+			var/list/L = ST.CardinalTurfs()
+			for(var/turf/T in L)
+				if(!istype(T, /turf/simulated))
+					continue
+				var/turf/simulated/S = T
+				S.wet = TRUE
+
+		if ("wall")										//Wall-tank: Wets it's own tile and the surrounding 8 tiles (3x3 square)
+			for(var/turf/T in view(src, 1))
+				if(!istype(T, /turf/simulated))
+					continue
+				var/turf/simulated/S = T
+				S.wet = TRUE
+
+
+//////////////////////////////			Note from FalseIncarnate:
+//		EXAMINE PROC		//			This proc is massive, messy, and probably could be handled better.
+//////////////////////////////			Feel free to try cleaning it up if you think of a better way to do it.
+
+/obj/machinery/fishtank/examine(mob/user)
+	..()
+	var/examine_message = list()
+	//Approximate water level
+
+	examine_message += "Water level: "
+
+	if(water_level == 0)
+		examine_message += "\The [src] is empty! "
+	else if(water_level < water_capacity * 0.1)
```

Comment:
May want to test this, I've a feeling if it's full it will print out every damn one of these when examining.

Reply:
It doesn't.

Reply:
>`else if`

That said, it should still be replaced with a `switch()`
`switch()` conditions have to be constant expressions, but you can just pass `water_level / water_capacity` instead of multiplying the proportions

Reply:
@Exxion Can you clarify ?
```switch(floor(10*water_level/water_capacity))
    if (0.1)
       ....```
Assuming BYOND has a floor command

Reply:
```
switch(water_level / water_capacity)
    if(0.1 to 0.25)
```

The DM Ref is pretty shitty, but you should still at least check it

Reply:
I r dumb
Thanks